Jc leo

jchanbroad opened the pull request:
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

rtitle committed:
Leo automation tests: Remove gs://leonardo-swat-test-bucket-do-not-delete (#468)

* Remove special gs://leonardo-swat-test-bucket-do-not-delete bucket from Leo automation tests

* Fix bucket upload issues

* More patience

* Fix assertion due to extension rename

* Rename extension test

jacmrob committed:
Jr build efficiency fix - skip tests (#471)

* hash tag all images

* jr-build-efficiency-fix

* fix

* fix docker command

* pass hash to jupyter biuld

* try

* debug

* misplaced quotes

* hash tag ui image as well

* preserve ability to set DOCKER_TAG; default it to git hash and also tag all images with branch

akarukappadath committed:
Allow users to configure autopause settings for cluster (#473)

* new branch

* updated swagger wording and uningnored test

aweng98 committed:
Retry op when getting 401 status from Sam (#475)

* retry sam 401

retry

fix

* missed one method

* samRetryTimeout 1 sec

vkumra-broad committed:
Issue 449: Leo not working in an iframe (#479)

akarukappadath committed:
Update jupyter notebook terminal to use bash instead of dash (#484)

* updated Jupyter notebook terminal to use bash

* reverting changes to run-jupyter-local.sh

rtitle committed:
Make cluster startup script backwards compatible (#482)

akarukappadath committed:
cache key no longer checks tokenExpiresIn value (#487)

kyuksel committed:
Synchronous portion of PUT createCluster v2 (#480)

kyuksel committed:
Update Hail_Tutorial.ipynb (#489)

rtitle committed:
Latest tensorflow is incompatible with our version of numpy (#505)

* Latest tensorflow is incompatible with our version of numpy

* Also ignore TF import warnings

rtitle committed:
Disable autoFreeze to match prod

akarukappadath committed:
updated dataprocdisabled to look at reason instead of message (#508)

rtitle committed:
Reuse pet token for all GCS validations instead of re-fetching token each time (#502)

* Reuse pet token for all GCS validations instead of re-fetching token each time

* FINE: retry 401s to GCS

* Oops, typo

* Still getting 401s, log token to help debug

* PR feedback

* Don't fail on 401s during object validation

vkumra-broad committed:
Issue 500 : Add js code to check jupyter status to prevent autofreeze if terminal is open (#506)

* add js code to ping jupyter status end point every 60s
* use pip install instead of apt-get

vkumra-broad committed:
Issue 384: Create script to enable download as pdf (#498)

* install texlive-xetex

abaumann committed:
#463: Added user install script for gatk ... (#478)

* Added user install script for gatk that can be used on cluster initialization.  Does not deal with any python packages that are needed for certain tools, only deals with the jars.

* Addressing PR review - changed hardcoded version and installed to /etc/ rather than /bin/

akarukappadath committed:
Added tidyverse to preinstalled R packages (#509)

* added tidyverse and test

* modified DF

* removed redundancies in R packages

* adding positive test

* different positive check

aweng98 committed:
Flaky test Jupyter kernel fix (#485)

* Kernel flaky fix

fix

fix

address feedback

feedback

wip

* upgrade workbench-service-lib

* 2 minutes in Hail test

* fix kernel ready

* feedback

* unmonitoring cluster delete in NotebookInstallSpec

* fix

akarukappadath committed:
Replaced gcloud with relevant subpackages and pegged their versions (#514)

* adding bigquery for testing

* removing redundant packages

* got rid of deprecated google-cloud

* spelling error

* added rudimentary test

* docker changes

* added tests for versions

* removed parts from tidyverse branch

* resolved test errors

* ignoring warnings

* finalizing

vkumra-broad committed:
Apt-get cleanup (#516)

* apt get clean up

vkumra-broad committed:
Issue  499: Update Leonardo Content Security Policy (#521)

Added content security policy to jupyter config

aweng98 committed:
Tagging some Leo tests as SmokeTest (#524)

* smoketests

* fix

* logback

aweng98 committed:
increase hail tutorial timeout (#526)

rtitle committed:
Use http mirror because of broken cert (#527)

kyuksel committed:
Async Cluster Creation - Season Finale! (#511)

akarukappadath committed:
Modularized cluster model so that more fields can be added to the model  (#522)

* modularized cluster model

* cleaning up some older stuff

* added tests

* removed unneccessary import statements

* increase hail tutorial timeout (#526)

* Use http mirror because of broken cert (#527)

* modularized cluster model

* cleaning up some older stuff

* added tests

* removed unneccessary import statements

* google auth

akarukappadath committed:
Fixed compilation errors from merge conflict (#529)

* fixed compilation errors

* fixed test compilation errors

rtitle committed:
Added wss://notebooks.firecloud.org:* to CSP (#532)

kyuksel committed:
Bug fix for cluster creation API v2  (#535)

* Fix bug in cluster creation API v2 causing GET 404s. Add more logging around cluster creation steps.

* Test cluster creation response time only for V2

akarukappadath committed:
Added defaultClientID to cluster  (#537)

* ready for manual testing

* fixed liquibase type issue

* liquibase mistake fixed

* console output for testing

* typo fix

* added defaultClientId to ClusterInitValues

* quote strugs

* added monitorDelete

* ready for manual testing

* fixed liquibase type issue

* liquibase mistake fixed

* console output for testing

* typo fix

* added defaultClientId to ClusterInitValues

* quote strugs

* added monitorDelete

* removed console log

* PR suggestions

* more PR fixes

* final

* typo fix

aweng98 committed:
3 jvm (#540)

akarukappadath committed:
Implemented makeCluster and save functions to reduce boilerplate in test code (#543)

* trying stuff

* updated one test

* ClusterComponentSpec updated

* more unit test updated

* all tests passed

* fixing some stray c#s

* PR edits

* fixed PR issues

* separated CommonTestData and TestComponent

jacmrob committed:
always push images to GCR (#550)

aweng98 committed:
Minor Leo test improvements (#551)

* increase BigQuery executeCell timeout to 5 min

* additional change

* latest workbench-lib

akarukappadath committed:
Finished updating tests based to reflect makeCluster method (#549)

* preliminary changes

* some merge conflicts

* clustername capital issue fixed

rtitle committed:
Remove 10s createCluster check (#553)

rtitle committed:
Correctly handle null stagingBucket in the Cluster response (#554)

kyuksel committed:
Example config update to set akka-http parsing max-content-length parameter (#556)

davidangb committed:
CORS support for X-App-Id header (#533)

* CORS support for X-App-Id header

* update CORS unit tests

aweng98 committed:
latest version (#560)

zarsky-broad committed:
Allow http/js nbextension (#561)

aweng98 committed:
Set default cluster monitorDelete = false (#576)

* set default monitorDelete = false

* fix

* fix

vkumra-broad committed:
install Ronaldo on jupyter docker, add automation test and include example use case (#575)

rtitle committed:
Enable auto pause (#584)

kyuksel committed:
Persist stopAfterCreation flag in the database (#579) (#586)

rtitle committed:
Async creation errors should error out the cluster (#592)

rtitle committed:
Retry docker-compose (#591)

* Retry docker-compose

* bash better

* Upgrade docker too

* PR feedback, haven't tested

* bash typo

* PR feedback

rtitle committed:
Implementation of zombie cluster detection (#587)

* Implementation of zombie cluster detection

* Minor tweaks

* Don't zombify creating clusters, and fix error message

* Also check the project's billing

* Update libs dependency for bugfix

* PR feedback

* Add unit test to make sure Google errors don't cause zombification

* Update wb-libs to non-SNAP

rtitle committed:
Don't continually monitor zombie clusters (#594)

* Don't continually monitor zombie clusters

* Remove unneccesary check

rtitle committed:
Add a generic bash retry function which actually works (#598)

* Add a generic bash retry function which actually works

* Remove unused variable

rtitle committed:
Increase selenium timeout for opening notebooks after resume (#600)

* Configure selenium timeout for with(New|Open)Notebook, increase it for the pause/resume tests

* Update DNS cache after the DB update to prevent race conditions

aweng98 committed:
Rework test "deleting an already deleting cluster should be a no-op" (#606)

* rework

* fix

rtitle committed:
Change jupyter to listen on localhost instead of * (#607)

rtitle committed:
Save jupyter.log to Jenkins after create and start cluster (#610)

* Save jupyter.log to Jenkins after create and start cluster

* Use .projectNameString

vkumra-broad committed:
Change default kernel to Python2 (#616)

* change default kernel to Python2

kyuksel committed:
Quick and dirty fix for init bucket creation failures under heavy load (#617)

vkumra-broad committed:
add retries to apt-get update (#618)

vkumra-broad committed:
Ignore Cross domain cookie auth test (#620)

* ignore test

rtitle committed:
List clusters by project (#623)

* List clusters by project

* Retry IOExceptions from Sam

* PR feedback

* Fix IOException retry condition

vkumra-broad committed:
save jupyter logs on cluster delete (#630)

* save jupyter logs on cluster delete

rtitle committed:
Fix node security vulnerabilities (#626)

MatthewBemis committed:
return 204 on success for setCookie endpoint (#633)

MatthewBemis committed:
#625: Make Dataproc init script timeout configurable (#628)

* add defaultExecutionTimeout from config

* set the timeout to 30 mins in reference.conf, can override in leo.conf

MatthewBemis committed:
#596: Return 401 only when it's an authentication error (#632)

* return 401s only when it's an authentication error

MatthewBemis committed:
#536: Delete cluster init buckets after cluster is deleted (#631)

* add call to delete init bucket in handleDeletedCluster

vkumra-broad committed:
Comment out the loc/deloc log file cleanup for localize tests (#642)

* comment out cleanup code

calbach committed:
Fix null check for unset googleClientId (#644)

Likely this was a regression with the addition of "defaultClientId". If a default is unspecified, `googleClientId` is now set to `""`. `"" == null` fails, so we never request the client ID from the parent frame.

akarukappadath committed:
Creating clusters cannot be deleted (#638)

* initial tries

* initial tries

* all unit tests passing

* all unit tests working

* automation tests trial 1

* automation tests trial 2

* added time out to eventually (trial 3)

* monitor create (trial 4)

* final touches

* cleaning up PR

rtitle committed:
Upgrade java to u191, the old one doesn't exist anymore (#647)

rtitle committed:
ignore non-deterministic test (#649)

* ignore non-deterministic test

* Fix mlr assertion

rtitle committed:
Experimental: obtain apt-key in init-actions.sh to work around interm… (#645)

* Experimental: obtain apt-key in init-actions.sh to work around intermittent apt authentication failures

* apt-key update for good measure

rtitle committed:
Increase akka patience in ClusterMonitorSpec (#651)

* Increase akka patience in ClusterMonitorSpec

* Represent configs better

rtitle committed:
Add AoU domains to CSP (#646)

vkumra-broad committed:
Issue 652 - Check file status before copying in the localize extension (#655)

davidangb committed:
update serviceTest version to get bugfixes (#658)

akarukappadath committed:
Updating PR Template  (#659)

* Update PULL_REQUEST_TEMPLATE.md

* Update PULL_REQUEST_TEMPLATE.md

rtitle committed:
Add: (#660)

https://stable.fake-research-aou.org
https://app.terra.bio

kyuksel committed:
Separate user script/extension testing from NotebookInstallSpec (#662)

rtitle committed:
bump merge to 1.2.1

rtitle committed:
Null check window.opener before using it (#667)

akarukappadath committed:
Retries for "Jupyter kernel is not ready" error (#668)

* trying retries

* cleaning up PR

* only retries if Jupyter kernel fails

* fixing compilation errors

rtitle committed:
Temp: ignore PySpark3 test until Docker upgrades are merged (#674)

vkumra-broad committed:
Refactor DNS Cache for horizontal scaling (#612)

MatthewBemis committed:
Increase timeout to wait for jupyter kernel after resuming cluster in tests (#675)

* Update kernel timeouts in ClusterMonitoringSpec.scala

* log time it took for kernel to get ready

vkumra-broad committed:
add retries to docker execs (#684)

akarukappadath committed:
(WIP) DockerExtinguisher  (#654)

* changed spark and debian

* fixed image version

* trying deb9

* changed version #

* reintroduced some hail things

* changed hail things to spark things

* Revert "changed hail things to spark things"

This reverts commit a551dac15e4e5df91ef4d30b81825bd6e5e0edc7.

"rolling back"

* Revert "reintroduced some hail things"

This reverts commit 409083013f0ea2a7724e1c2e69edfd0248ee38e6.

"rolling back more"

* uncommented line that should not have been commented

* recommenting

* docker FIRE

* Dockerfile fixes, it builds now

* clean up

* uncommenting packages

* commenting things out

* R and others

* going back on R version

* working on workers

* well that was stupid

* everything workinggit add .

* removing R for now

* what R u doin

* R is ok again

* well that was also stupid

* working on tests

* envs

* pyspark

* fixing tests

* fixed str not bytes

* what is going on

* fixed pyspark2

* pyspark things

* trying open wb

* hail things

* hail please

* removed summarize for now

* replaced hail tutorial

* IS IT OVER

* testing for ggplot

* ITS DONE WHAT

* NOW ITS ACTUALLY DONE YEAH

* updated tornado

* going back to older tornado because of backwards compatibility issues

MatthewBemis committed:
#288 + #496: PATCH cluster endpoint: Resize cluster and update autopause threshold (#669)

*  patch cluster endpoint to update num workers and autopause threshold

rtitle committed:
[692] Allow users to pick their Jupyter image at cluster creation time (#693)

* Make Jupyter image configurable

* Reduce joins in mimimalClusterQuery; code clean up

* Add a timestamp column to CLUSTER_IMAGE

MatthewBemis committed:
fix broken swagger yaml (#700)

* fix broken swagger yaml

ansingh7115 committed:
update workbench-service-test version (#701)

rtitle committed:
Log time it takes to do the Hail import and bump timeout to 10 min (#698)

* Log time it takes to do the Hail import and bump timeout to 10 min

* Fix merge issue

* Unignore tests and remove flake

* Fix assertions

rtitle committed:
Automation tests: add jitter to project claim (#706)

akarukappadath committed:
Make cluster scopes definable by app/user (#705)

* first try

* fixed comp error

* actually fixed all the comp errors this time

* scopes for preexisting clusters

* was this really a semicolon issue

* nah it was something even stupider

* still not working

* made scopes a set

* fiab time

* updated automation test models

* automated test

* everything working

* merging

* everything working after rebase

* cleaning PR

* actually yaml

* YAAAAML

* get rid of dsub

RLuckom committed:
Rl version endpoint databiosphere (#711)

* add version endpoint

* update test

rtitle committed:
RStudio Phase 1 (#690)

* Add rstudio image and update build scripts

* Remove jupyter-docker directory

* Add rserver.conf and listen on port 8001

* Dockerfire changes didn't carry over in git mv

* Clean up

* 1 more cleanup

* Update rstudio docker and add some README's

* Fix image name in build-tool.sh

* bash fix

* Better scheme for backwards compatible jupyter images

* quote bash vars

ansingh7115 committed:
GAWB-3999: Fail billing project creation if gpalloc runs out of billing projects (#715)

* make tests fail if gpalloc has no projects to allocate

* real hash

MatthewBemis committed:
Ignore mysteriously failing PDF download test (#717)

Not crucial functionality, so we can ignore for now

rtitle committed:
[710] Fix broken gfortran dependency (#719)

* [710] Fix broken gfortran dependency

* docker syntax

* fix assertion

aweng98 committed:
browser timeout due to long inactivity (#722)

kyuksel committed:
Refactor cluster monitoring for horizontal scaling (#689)

* Scheduled task for database scrubbing of cluster statuses
* remove references to clustermonitorsupervisor from leo service
* remove recreate when deleting
* add code to remove cluster from the supervisor monitor

rtitle committed:
Just compare clusters by id (#723)

* Just compare clusters by id

* access actor state outside the Future

rtitle committed:
Don't delete Creating clusters in the ClusterMonitorActor (#725)

davidangb committed:
update service-test to adopt billing scope change (#720)

update service-test to adopt billing scope change

vkumra-broad committed:
Horizontal Scaling : checkCluster should not use full cluster query (#726)

* use min query for clusters

vkumra-broad committed:
Issue 564 : Seperate front and back leo (#724)

* Separate front and back leo

vkumra-broad committed:
Issue 404: Install TOC extension by default (#721)

* install TOC extension and nbconfigurator by default

rtitle committed:
Exponentially retry 409s from GoogleIamDAO policy changes, to support concurrent cluster creations in the same project (#737)

rtitle committed:
no op

akarukappadath committed:
JupyterLab Extension Support (#714)

* fiab time

* fixed test

* right node and all

* YAY

* lab ext file

* cleaning up PR

* testing other input files

* newest juplab

* extracting in bash fine

* trying again

* all donegit add .

* all done

* more docs

calbach committed:
Update run-jupyter-local.sh post templating changes (#665)

* Update run-jupyter-local.sh post templating changes

* fix paths

rtitle committed:
Build python3.6 from source (#748)

* Build python3 from source

* path change

* Update Dockerfile

* Fix python install path

* More sane init-actions.sh

* Add ldconfig

rtitle committed:
Make dataproc zone configurable (#760)

vkumra-broad committed:
Issue 733: If google marks cluster as Error but doesn't return error code then don't continue to monitor the cluster (#764)

* fix clustermonitoractor

akarukappadath committed:
Bash script for hotswapping leo jar on fiab (#766)

* starting

* still trying

* hotswap.sh in automation

* changed permissions

* cleaned up

* comments and error

vkumra-broad committed:
Issue 751 : Install gcsfuse on clusters by default (#765)

* install gcsfuse

akarukappadath committed:
Recursively copy jupyterlab extension (#769)

* -r

* documentation update

aweng98 committed:
retry on chrome launch error (#772)

MatthewBemis committed:
upgrade Docker version to 18.06.2 (#773)

* upgrade Docker to 18.06.2~ce~3-0 since it's stable

rtitle committed:
More efficient database unmarshaling (#775)

* More efficient database unmarshaling

* Update workbenchGoogle to the latest

akarukappadath committed:
Specify JupyterLab extension from single JS file  (#778)

* first try

* yay

* doc update

* automation test

* fixed test

* ts support

akarukappadath committed:
trying SparkR 2.2.3 (#781)

MatthewBemis committed:
Speculative fix: discard entity bytes when request fails to solve Apache CPU spikes (#786)

* discard entity bytes when request fails

rtitle committed:
Specify dataproc minor version to peg Spark at 2.2.3 (#789)

* Specify dataproc minor version to peg Spark at 2.2.3

* Revert "Specify dataproc minor version to peg Spark at 2.2.3"

This reverts commit 4ba5393e01968b74d9b4c489b51f9f2337d78a40.

* Ignore test

rtitle committed:
Test fix: load library in a different cell (#790)

rtitle committed:
[770] Token refresh not working from JupyterLab (#787)

* Enable google_sign_in.js as a lab extension

* Use common google_sign_in.js between jupyter and Lab

* fix init-actions.sh

* Add comment

* All extension installation happens as root; lab probably needs to as well

vkumra-broad committed:
Issue 757 : Maintain state in RStudio (#767)

* add rstudio setting to persist session state.

MatthewBemis committed:
RStudio phase3: genericize proxy routes and deprecate old ones (#718)

* rstudio phase 3

rtitle committed:
Fix ggplot on python 2 (#798)

* Add test for ggplot fix

* Fix ggplot issue for py2.7

* Remove extra RUN label

* dist-packages for py2

MatthewBemis committed:
#801: Install python dev libs in RStudio container (#799)

* experimental: install python dev libs in RStudio container

* improvements + install pip by default

rtitle committed:
Upgrade hail to the latest 0.2 build (#800)

* Upgrade hail to the latest 0.2 build

* use different hail zip hash

* Hail zip is obsolete

MatthewBemis committed:
minor cleanup, create docker volume for rstudio (#803)

* minor cleanup, create docker volume for rstudio

MatthewBemis committed:
Stop installing gcsfuse on clusters (#810)

rtitle committed:
Upgrade jackson to 2.9.8 to address CVE reports (#812)

akarukappadath committed:
Set staging bucket to be deleted a specified number of days after deletion of cluster (#792)

* works?

* Well that was silly

* lets see

* debugging

* what

* catching up

* one more unit test to go

* wat

* unit tests are good

* cleaning up PR

* no magic numbers

* some cleaning up

* well that was silly

* comment update

akarukappadath committed:
Check that all specified cluster tools are ready before setting cluster to Running from Starting (#809)

* maybe works

* working on tests

* done

* cleaning up

* removed join

* more scala-esque

* automation test

* automation test

akarukappadath committed:
Automated Test for JupyterLab and Test Refactor (#749)

* progress

* pls

* yes

* restructuring

* refactored

* for alex

* more for alex

* test working

* refactoring

* lab error

* refactored

* fixed errors

* why

* started lab loc/deloc

* PR fixes

* couple more fixes

* fixed errors

* remove toc.tgz

* PR fixes

* DONE WOW

* merges are messy

akarukappadath committed:
fixed (#830)

akarukappadath committed:
done (#828)

vkumra-broad committed:
update ronaldo example notebook (#833)

akarukappadath committed:
Jupyterlab Extension Install Fix (#835)

* bask to jupyter user

* done

* added auto test

* removing print statement

* permission error resolved

helgridly committed:
sbt batch mode so it doesn't hang while building (#837)

rtitle committed:
Upgrade hail to 0.2.11 and improve tests (#836)

* Upgrade hail to 0.2.11 and improve tests

* Add escape

* Bump the hash again

rtitle committed:
Try using keyserver.ubuntu.com (#845)

* Try using keyserver.ubuntu.com

* Make token revoke optional

rtitle committed:
Set workspace name/bucket in environment variable (#841)

* First pass: set workspace name and bucket in EVs

* Do the bootstrapping in bash

* Minor fixes

* Fix sed command

* Another sed fix

* Fix tests: new folder stuff wasn't working

vkumra-broad committed:
add retries to docker compose and python installation (#853)

akarukappadath committed:
Terminal timeout after 2 mins (#851)

* changing timeout for testing

* Update google_sign_in.js

akarukappadath committed:
List clusters stale results fix (#849)

* first try

* clusters created by user

* fixed failing test

* Update google_sign_in.js

* Update SamAuthProvider.scala

* removed unnecessary db query

* unit test

* well that was stupid

* test struggles

vkumra-broad committed:
PR to fix leo automation test failures (#861)

vkumra-broad committed:
Change fix cat cmd for docker compose files in Init-actions (#866)

rtitle committed:
[864] WORKSPACE_NAME environment variable is empty when workspace name has spaces (#867)

MatthewBemis committed:
update swagger to document how cluster scope configuration works (#872)

Qi77Qi committed:
upgrade some dependencies (#814)

* upgrade some dependencies

* enable debugging

* revert logback change

* update travis file

* add some more scalac options

* remove unused imports

* disable unused warning in console

* remove dns setting

* revert akka-http version

MatthewBemis committed:
remove colon from yaml (#877)

rtitle committed:
Expand scope of cluster PATCH endpoint (#855)

* First pass: support changing master machine type and disk in PATCH request

* add unit tests

* Fix compilation error

* update swagger and fix tests

* Add auto test for cluster patch

* Add TODOs based on testing

* Handle changing machine type gracefully

* Make Stopped an updatable status

* Fix auto test, hopefully

* Fix assertion

* I think it should work now

* Fix auto test

* Post rebase fixes

* PR feedback

rtitle committed:
Bugfix: disabling auto-pause does not work (#882)

* Disabling auto-pause does not work

* Forgot .save()

* Don't auto-pause if autopauseThreshold is zero

akarukappadath committed:
Install GATK, CNN libraries, samtools, and OpenJDK in default dockerfile (#865)

* works

* added test

* Update Dockerfile

* java update

* fixing test

* cleaning tests

* test fixes

* Update NotebookInstallSpec.scala

* Update Dockerfile

* Update NotebookTestUtils.scala

Qi77Qi committed:
check kernel for auto freeze (#874)

* check kernel for auto freeze

enable debugging

revert logback change

update travis file

check kernel for auto freeze

* add unit test

* make test better

* add logging

Qi77Qi committed:
allow user to pass properties when creating cluster (#871)

* pass properties

enable debugging

revert logback change

update travis file

* allow user to pass properties when creating cluster

add some validation on properties

* revert akka-http version

* address comments

* deal with null value

* one more test

jchanbroad committed:
pushing the new test file

rtitle committed:
Added missing Terra domains to Content-Security-Policy (#893)

jchanbroad committed:
added withwebdriver to test localize

akarukappadath committed:
Possible fix for terminal time out: changed ping interval (#896)

* changed ping interval

* Update jupyter_notebook_config.py

jchanbroad committed:
all tests pass

jchanbroad committed:
reduced runtime of test

Qi77Qi committed:
umm, why didn't I add the test before (#902)

Qi77Qi committed:
add more logging (#904)

rtitle committed:
Don't overwrite Dataproc properties (#907)

akarukappadath committed:
DockRfix (#900)

* trying

* unpin r version

* potential fix

* yes

* Update Dockerfile

* more packages

* Update Dockerfile

* still not working

* try something

* builds

* fixed?

* cleaning up

* changed timeouts and added tidyverse

* cleanup

jchanbroad committed:
with the commented parts

akarukappadath committed:
done (#899)

Qi77Qi committed:
add one more unit test (#908)

rtitle committed:
gcc is broken in the Jupyter image (#892)

* Can R run with the old gcc?

* Add auto test for gcc

* Another approach

* Docker fix

jchanbroad committed:
latest without createNewCluster

Qi77Qi committed:
lint cluster name before create cluster with dataproc (#895)

* lint cluster name before create cluster with dataproc

* use NoStackTrace

akarukappadath committed:
Change content security policy to support IGV notebook extension (#917)

* done

* added userscript

jchanbroad committed:
works now

jchanbroad committed:
test for jenkins

jdcanas committed:
Jc creation zombie (#901)

* add functionality and config to consider clusters hanging in created as zombies. INIT SCRIPT HAS AN HOUR LONG SLEEP FOR FIAB TESTING.

* refactoring tests and cleaning up comments

* adding an additional test

* fixing isActiveInGoogle logic and cleaning up comments

jchanbroad committed:
combined creating new cluster and hard-coded cluster

jdcanas committed:
Jc update cluster endpoint 400 patch (#936)

* add functioanlity and config to consider clusters hanging in created as zombies. INIT SCRIPT HAS AN HOUR LONG SLEEP FOR FIAB TESTING. (squash update: this has been reverted)

* refactoring tests and cleaning up comments

* adding an additional test

* adding initial tests and debug code for update service

* adding debug logging message

* adding possible fix for when workers is 0 and leo gives improper error

* fix for numWorkers = 0 and numPreemptibleWorkers = 0 causing 400

* reverting test code after experiment

* adding a test and refactoring nested if to a match

jdcanas committed:
Jc error message change (#939)

* add functioanlity and config to consider clusters hanging in created as zombies. INIT SCRIPT HAS AN HOUR LONG SLEEP FOR FIAB TESTING.

* refactoring tests and cleaning up comments

* adding an additional test

* adding initial tests and debug code for update service

* adding debug logging message

* adding possible fix for when workers is 0 and leo gives improper error

* fix for numWorkers = 0 and numPreemptibleWorkers = 0 causing 500

* reverting test code after experiment

* adding a test and refactoring nested if to match

* making error message less abrasive to user

* changes status code to something more appropriate

* adding swagger documentation and changing error code to be more descriptive

Qi77Qi committed:
fix indentation (#941)

jchanbroad committed:
commented out notebooksCanaryProject

jchanbroad committed:
changed application confignotebooksCanaryProject string

vkumra-broad committed:
Fix automation test failure (#943)

* fix automation test

jchanbroad committed:
pushing the createnewcluster code and commented the others out

using with new cluster method

connection failed error

saving changes before switching to dev

jmthibault79 committed:
[IA-958 and IA-959][risk=no]Safe Mode extension (#945)

* add IA-979 comments

akarukappadath committed:
Allow users to see output from a failed userscript (#916)

* trying

* in process

* starting

* added piping

* fixed unit test

* fixed unit test

* init actions

* removed retries for user script

* Update LeonardoModel.scala

* PR updates

* for fiab

* updates for Qi's workbench-libs update

* removed debugging statements

* clusterSupervisorSpec update

* version for workbench libs

* unit tests passing

* clustermonitorspec

* that was stupid

* added test

* fixed error problem

* fixed automation compilation error

* whats going on

* fixed bigquery test

* Update Boot.scala

* specifying return type

* oops

* moved around

* PR fixes

jchanbroad committed:
notebooks canary test good for pr

rtitle committed:
Remove Leo whitelist config setting (#949)

* Remove Leo whitelist config setting

* Remove obsolete tests

MatthewBemis commented on automation/src/test/scala/org/broadinstitute/dsde/workbench/NotebooksCanaryTest.scala:
```diff
@@ -0,0 +1,50 @@
+        withWebDriver { implicit driver =>
+          withNewNotebook(cluster, Python3) { notebook =>
+            print(notebook.executeCell("""print("hi")""") shouldBe (Some("hi")))
+            println(clusterStatus shouldBe(ClusterStatus.Running))
```
i think the `println` here should be removed

MatthewBemis commented on automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoConfig.scala:
```diff
@@ -8,6 +8,8 @@ object LeonardoConfig extends CommonConfig {
     val apiUrl: String = leonardo.getString("apiUrl")
     val notebooksServiceAccountEmail: String = leonardo.getString("notebooksServiceAccountEmail")
+    val notebooksCanaryProject: String = leonardo.getString("notebooksCanaryProject")
+    //val notebooksCanaryProjectProd: String = leonardo.getString("notebooksCanaryProjectProd")
```
is this meant to be commented out?

MatthewBemis commented on automation/src/test/scala/org/broadinstitute/dsde/workbench/NotebooksCanaryTest.scala:
```diff
@@ -0,0 +1,50 @@
+
+
+  implicit val authToken: AuthToken = ronAuthToken
+  "Test for creating a cluster and localizing a notebook" - {
```
nit: i wonder if this should be called something like "Leonardo Canary Test" and the following block, which contains the actual test, can be the description of what should happen

MatthewBemis reviewed (commented):
just a few minor comments! overall looks good

jchanbroad committed:
not working but added bash script execution

MatthewBemis committed:
add datastage.terra.bio (#951)

jmthibault79 committed:
[IA-951][IA-952][risk=low] Deploy Welder to clusters but do not enable it(#948)

- install welder on leo cluster
- fix unit test errors
- upload welder-docker-compose to init bucket
- leo clusters install welder on creation
- set temp welder image
- Use the existing /work dir as the shared dir
- welder enabled flag

Qi77Qi committed:
remove unmaintained code (#957)

MatthewBemis committed:
Run the Jupyter user script as privileged (#956)

jchanbroad committed:
trying this for dsl seed job

jchanbroad committed:
for dsl seed job

MatthewBemis committed:
Update README.md

jchanbroad committed:
added settings in build.sbt

jchanbroad committed:
minor changes to import statemnets

jchanbroad committed:
assigned testScript variable

jchanbroad committed:
changed directory of shell script

jchanbroad committed:
removed val

jchanbroad committed:
added path to script

jmthibault79 committed:
[risk=low]Deploy Jupyter safe mode plugin onto clusters (#953)

jmthibault79 committed:
[risk=no][IA-1026]Defer the execution of Welder until after its container is created (#959)

* Start Welder via init-script instead of docker-compose, to allow for ADC
* update welder hash to 0b28747
* Add comments linking to enable-welder story

helgridly committed:
leo requires java 8 (#963)

jchanbroad committed:
removed automation/

jchanbroad committed:
added class path

jchanbroad committed:
added path firecloud-automated-testing

jchanbroad committed:
add pwd

jchanbroad committed:
put import statement

jchanbroad committed:
commented out pwd

jchanbroad committed:
retry

jchanbroad committed:
added pwd back in again

jchanbroad committed:
renamed vals

jchanbroad committed:
commented bash script execution out

jchanbroad committed:
added ls and pwd

jchanbroad committed:
commented canary out

jchanbroad committed:
changed clusterTimeRes to time

jchanbroad committed:
added grep

jchanbroad committed:
added grep time

calbach committed:
Update work mount location to match Jupyter home (#962)

* Notebook dir with updated /localize

* Fix README

* Only change notebook root for Welder VMs

vkumra-broad committed:
Change getCluster to listCluster in cluster create (#964)

* change get to list

* pr feedback

MatthewBemis committed:
create init bucket in user's project (#950)

* create init bucket in user's project

* createBucket actually takes care of default obj acls

* Add retries to both bucket creations

jmthibault79 committed:
[no JIRA][risk=no]Rename addClusterLabels to augmentClusterRequest because it does more than labels (#952)

- split out the labeling to a new addClusterLabels
- refactor augmentClusterRequest

MatthewBemis committed:
allow all terra.bio domains via wildcard in csp (#966)

MatthewBemis committed:
ignore LabSpec test until lab is enabled again (#969)

jdcanas committed:
Jc jupyter extension (#960)

* Deploy Jupyter safe mode plugin onto clusters

* safety first

* copy into container with init.sh

* Very important to match names!

* initial commit with jupyter locking extension functioning as expected in isolation

* removing notification banner from Joel's ui extension; my extension absorbed that functionality

* cleanup safe mode extension

* cleaning up some comments

* changes made to incorporate into docker

* PR feedback

* small change to edit mode modals

* adding proper links with templating

* updating safe-mode to call check meta properly via templating values

* updating file with PR feedback

* pr feedback

* disabling plugins by default

* updating init functions to print

* adding copy into gcs of new nbextensions:

* adding small fix

* updating safe mode with a few fixes

* adjusting google project and cluster variable  scoping

* small changes to help with templating

* touching up some comments

* potential fix for clobbering of templated resource

* adding anticipated lock endpoint name

* touching up requests

* touching up urls used

* fixing up console.logs and async/awaits

* renaming for consistency

* remove TODOs

* safe mode and copying transitions handle file creation differently. meta state removed

* remove unecessary var

* removing local testing changes

* cleanup and additional PR feedback

* small fixes based on fiab testing

* adding readme and making small safe mode change

* comment touching up

calbach committed:
Implement initial Welder server extension (#968)

jchanbroad committed:
ready for pr

jchanbroad committed:
reuploading

jchanbroad committed:
removed line

calbach committed:
Welder ext: handle deletes and errors (#971)

helgridly committed:
Use high-security networks when requested by project (#965)

jchanbroad committed:
removed prod from leoconfig

jchanbroad commented:
jenkins rebuild

jchanbroad commented:
jenkins retest

jchanbroad committed:
added tag

jchanbroad commented:
jenkins rebuild

jchanbroad commented:
jenkins retest

jchanbroad commented:
jenkins retest

rtitle committed:
Update wb-libs google hash for nullcheck fix (#973)

* Update wb-libs google hash for nullcheck fix

* Use non-SNAP version

jchanbroad commented:
jenkins retest

jchanbroad commented:
travis retest

jchanbroad committed:
added tag

akarukappadath committed:
Flag to enable welder per cluster (#970)

* added welder check for google project

* first pass

* compilation error fixes

* using latest welder

* fiab struggs

* Revert "fiab struggs"

This reverts commit 5db2a045c10686ae6e92e600165c8f7145ed1580.
pls

* some PR fixes

* unit test fix

* back 2 the Future

* no more throwing

* removed check

* docs update

* removing more check things

jchanbroad commented:
jenkins retest

jchanbroad committed:
removed tag

jchanbroad committed:
trying this now

calbach committed:
Export WELDER_ENABLED for docker-compose (#975)

calbach committed:
Fix swagger syntax errors (#974)

jchanbroad committed:
ready for pr

rtitle committed:
[IA-1086] Reconcile shared Welder volume ownership/permissions (#977)

* Change welder hash

* Fix welder startup script

* Set umask and propagate welder uid

* comment tweak

* Improve comment

* Update welder hash

jchanbroad committed:
added a tag

jchanbroad committed:
changed billingproject assignment

jchanbroad committed:
Delete canary_events.json

andy7i commented:
jenkins retest

jdcanas commented:
Closing this as Jocelyn's co-op has ended and this has diverged pretty far. There is a (currently unprioritized) ticket here for its continuation: https://broadworkbench.atlassian.net/browse/IA-1271
